FEM/rf_pcs.cpp

```diff
@@ -8049,8 +8052,21 @@ double CRFProcess::evaluteSwitchBC(CBoundaryCondition const & bc, CBoundaryCondi
 
 					for (std::size_t temp_i(0); temp_i < m_st->getNumberOfConstrainedSTs(); temp_i++)
 					{
-						if (st_vector[cnodev->getSTVectorGroup()]->isCompleteConstrainST(temp_i)
+						int st_conditionst = (st_vector[cnodev->getSTVectorGroup()]->isCompleteConstrainST(temp_i)
 							&& st_vector[cnodev->getSTVectorGroup()]->getCompleteConstrainedSTStateOff(temp_i))
+							? 1 : 0;
+
+#if defined(USE_MPI) || defined (USE_PETSC)
+						int int_buff = 0;
+#ifdef USE_PETSC
+						MPI_Comm comm = MPI_COMM_WORLD;
+#else
+						MPI_Comm comm = comm_DDC;
+#endif
+						MPI_Allreduce(&st_conditionst, &int_buff, 1, MPI_INT, MPI_SUM, comm);
```

Comment:
@waltherm collective communication function can’t be called here because that not call partitions satisfy the condition of (line 8044)
```
 			if (st_node.size() > 0 && (long) st_node.size() > i)
``` 
The means some compute cores skip the if-condition, while the other move inside the if-condition to call ``MPI_Allreduce``, which leads to a dead lock here.

Due to this reason, I have to close this PR and want to discuss the issue with you later.